spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java

```diff
@@ -88,10 +89,10 @@ public AetherModuleResolver(File localRepository, Map<String, String> remoteRepo
 		}
 		this.localRepository = localRepository;
 		this.remoteRepositories = new LinkedList<>();
-		if (!CollectionUtils.isEmpty(remoteRepositories)) {
-			for (Map.Entry<String, String> remoteRepo : remoteRepositories.entrySet()) {
-				RemoteRepository remoteRepository = new RemoteRepository.Builder(remoteRepo.getKey(),
-						DEFAULT_CONTENT_TYPE, remoteRepo.getValue()).build();
+		if (!CollectionUtils.isEmpty(remoteRepositoriesSet)) {
+			for (String remoteRepo : remoteRepositoriesSet) {
+				RemoteRepository remoteRepository = new RemoteRepository.Builder("",
```

Comment:
Please see my comment above. I think we can avoid a lot of complexity by using a Collection type directly in the ConfigurationProperties class. Based on this discussion about "id", it should probably be a Map.